ext/intl/common/common_date.cpp

```diff
@@ -137,7 +139,8 @@ U_CFUNC int intl_datetime_decompose(zval *z, double *millis, TimeZone **tz,
 			return FAILURE;
 		}
 
-		*millis = U_MILLIS_PER_SECOND * (double)Z_LVAL(retval);
+		datetime = Z_PHPDATE_P(z);
+		*millis = U_MILLIS_PER_SECOND * ((double)Z_LVAL(retval) + datetime->time->f);
```

Comment:
I don't think getTimestamp should have been call as a PHP function at all here. THen again, timelib should have a function for returning timestamp/fraction, and all the other fields really.